### ISSUE TITLE
Pipingbag\Routing\Filter\ControllerFactoryFilter::beforeDispatch missing?

### ISSUE THREAD

**Issue by btaens** [lorenzo/piping-bag#10]
It seems beforeDispatch is missing from the class mentioned in the title.

I'm not sure if I'm doing it wrong, and it's supposed to be this way, but because of this the controller's $request->params is never populated, so I can't find out its' _ext (extension) property, for example, if it was called as an html or json, or something.

**Comment by btaens** [comment]
Wait, I'm a moron. It's extending ParentFactory. It's still not populating the params though.

**Comment by lorenzo** [comment]
As I told you in the other ticket, you need the latest cakephp in the master branch. Once cakephp 3.0.3 is released, you can require the stable version again :)

**Comment by btaens** [comment]
I was using dev-master.

I have found the problem, I was trying to call request on initialize().
It's not exactly a bug, but perhaps it's worth noting in the docs somewhere that because the request object is set after class initialization, request parameters are not accessible at startup (as initialize() gets called during __construct), therefore such operations should be done in beforeFilter(), and not at initialize().
Though now that I encountered this I realized that's bad practice anyway.
